packages/discord.js/src/structures/InteractionCollector.js

```diff
@@ -150,7 +150,13 @@ class InteractionCollector extends Collector {
     if (this.interactionType && interaction.type !== this.interactionType) return null;
     if (this.componentType && interaction.componentType !== this.componentType) return null;
     if (this.messageId && interaction.message?.id !== this.messageId) return null;
-    if (this.messageInteractionId && interaction.message?.interaction?.id !== this.messageInteractionId) return null;
+    if (
+      this.messageInteractionId &&
+      interaction.message?.interaction?.id &&
+      interaction.message?.interaction?.id !== this.messageInteractionId
```

Comment:
Seems like this change got rolled back?
```suggestion
      interaction.message.interaction.id !== this.messageInteractionId
```